tests/preprocessing/test_clustering.py

```diff
@@ -18,3 +19,34 @@ def test_opt_louvain(adata_neighbors):
     LOGGER.info(f"max resolution: {res_max}, max score: {score_max}")
     assert res_max == 0.7
     assert_near_exact(score_max, 0.7432787576640969, diff=1e-3)
+
+
+def test_cluster_optimal_resolution_louvain(adata_neighbors):
+    res_max, score_max, score_all = scib.cl.cluster_optimal_resolution(
+        adata_neighbors,
+        label_key="celltype",
+        cluster_key="cluster",
+        cluster_function=sc.tl.louvain,
```

Comment:
we should test with leiden now i think... as louvain is being deprecated as i understand.